explorer/test/Test/Pos/Explorer/Web/ServerSpec.hs

```diff
@@ -1,17 +1,264 @@
+import           Test.Hspec.QuickCheck        (modifyMaxSuccess, prop)
+import           Test.QuickCheck              (arbitrary, counterexample, forAll, (==>))
+import           Test.QuickCheck.Monadic      (assert, monadicIO, run)
+
+import           Pos.Arbitrary.Block          ()
+import           Pos.Block.Core               (Block)
+import qualified Pos.Communication            ()
+import           Pos.Core                     (EpochIndex (..), HasConfiguration,
+                                               LocalSlotIndex (..), SlotId (..))
+import           Pos.DB.Block                 (MonadBlockDB)
+import           Pos.Explorer.ExplorerMode    (runExplorerTestMode)
+import           Pos.Explorer.ExtraContext    (ExplorerMockMode (..), ExtraContext (..),
+                                               makeExtraCtx, makeMockExtraCtx)
+import           Pos.Explorer.TestUtil        (basicBlockGenericUnsafe, emptyBlk,
+                                               leftToCounter)
+import           Pos.Explorer.Web.ClientTypes (CBlockEntry)
+import           Pos.Explorer.Web.Server      (getBlocksLastPage, getBlocksPage,
+                                               getBlocksPagesTotal, getBlocksTotal,
+                                               pureGetBlocksPagesTotal,
+                                               pureGetBlocksTotal)
+import           Pos.Launcher.Configuration   (HasConfigurations)
+import           Test.Pos.Util                (withDefConfigurations)
+
+
+
+----------------------------------------------------------------
+-- Spec
+----------------------------------------------------------------
 
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+-- sack test cardano-sl-explorer --fast --test-arguments "-m Test.Pos.Explorer.Web.ServerSpec"
 spec :: Spec
-spec = describe "Test" $ do
-    it "should be equal" $
-        True `shouldBe` True
+spec = withDefConfigurations $ do
+    describe "Pos.Explorer.Web.Server" $ do
+        blocksPureTotalSpec
+        blocksPagesPureTotalSpec
+
+        blocksTotalUnitSpec
+        blocksPagesTotalUnitSpec
+        blocksPageUnitSpec
+        blocksLastPageUnitSpec
+
+        blocksTotalFunctionalSpec
+
+-- | A spec with the following test invariant. If a block is generated, there is no way
+-- that blocksTotal could be less than 1.
+blocksPureTotalSpec :: HasConfigurations => Spec
+blocksPureTotalSpec =
+    describe "pureGetBlocksTotal"
+    $ modifyMaxSuccess (const 200) $ do
+        prop "created blocks means block size > 0" $ emptyBlk $ \blk0 -> leftToCounter blk0 $ \blk ->
+            let mainBlock   = Right blk
+                blocksTotal = pureGetBlocksTotal mainBlock
+            in counterexample ("Total blocks sizes: " <> show blocksTotal <>
+                               "\n\nBlock: " <> show blk) $
+                 blocksTotal > 0
+
+-- | A spec with the simple test that two equal algorithms should work the same.
```

Comment:
Description is too general. When I read it I thought that its testing equality of any two given algorithms.

Reply:
Well, it is. But that just proves that both are correct (or equally wrong). Will improve.